onnxruntime/contrib_ops/cpu/skip_layer_norm.cc

```diff
@@ -138,26 +125,18 @@ void ComputeJob(
   IAllocatorUniquePtr<float> input_float_uptr = IAllocator::MakeUniquePtr<float>(alloc, num_elems);
   MlasConvertHalfToFloatBuffer(p_input, input_float_uptr.get(), num_elems);
 
-  if (!skip_float_uptr) {
-    skip_float_uptr = IAllocator::MakeUniquePtr<float>(alloc, num_elems);
-    MlasConvertHalfToFloatBuffer(p_skip, skip_float_uptr.get(), num_elems);
-  }
-
-  if (bias_data && !bias_float_uptr) {
-    bias_float_uptr = IAllocator::MakeUniquePtr<float>(alloc, num_elems);
-    MlasConvertHalfToFloatBuffer(bias_data, bias_float_uptr.get(), num_elems);
+  if (should_convert_skip) {
+    MlasConvertHalfToFloatBuffer(p_skip, skip_float_ptr, num_elems);
   }
 
   IAllocatorUniquePtr<float> output_float_uptr = IAllocator::MakeUniquePtr<float>(alloc, num_elems);
```

Comment:
I think in general, allocating buffers outside this function would be a better idea because it will has less calls of allocation.

Do you allocate the buffer inside `ComputeJob` intentially?